cloud/azure/deploy/website.go

```diff
@@ -0,0 +1,522 @@
+		contentType := mime.TypeByExtension(filepath.Ext(path))
+		if contentType == "" {
+			contentType = "application/octet-stream"
+		}
+
+		// Generate the object key to include the folder structure
+		var objectKey string
+
+		filePath := path[len(cleanedPath):]
+
+		// If the base path is not the root, include it in the object key
+		if config.BasePath == "/" {
+			objectKey = filepath.ToSlash(filePath)
+		} else {
+			objectKey = filepath.ToSlash(filepath.Join(config.BasePath, filePath))
+		}
+
+		name := strings.TrimPrefix(objectKey, "/")
+
+		existingMd5Output := pulumi.All(p.ResourceGroup.Name, p.StorageAccount.Name, p.website.ContainerName).ApplyT(func(args []interface{}) (string, error) {
+			resourceGroupName := args[0].(string)
+			accountName := args[1].(string)
+			containerName := args[2].(string)
+
+			serviceClient, err := getBlobServiceClient(p.ClientConfig.SubscriptionId, resourceGroupName, accountName)
+			if err != nil {
+				return "", err
+			}
+
+			existingMd5, err := getBlobMD5(serviceClient, containerName, name)
+			if err != nil {
+				return "", err
+			}
+
+			return existingMd5, nil
+		}).(pulumi.StringOutput)
+
+		blob, err := storage.NewBlob(ctx, name, &storage.BlobArgs{
+			ResourceGroupName: p.ResourceGroup.Name,
+			AccountName:       p.StorageAccount.Name,
+			ContainerName:     p.website.ContainerName,
+			Source:            pulumi.NewFileAsset(path),
+			ContentType:       pulumi.String(contentType),
+		}, opts...)
+		if err != nil {
+			return err
+		}
+
+		keyToInvalidate := pulumi.All(blob.ContentMd5, existingMd5Output).ApplyT(func(args []any) (string, error) {
+			newMd5, newMd5Ok := args[0].(*string)
+			existingMd5, existingMd5Ok := args[1].(string)
+
+			if !newMd5Ok || !existingMd5Ok {
+				return "", fmt.Errorf("failed to assert md5 types")
+			}
+
+			// if an existing ETag is present and it is different from the new ETag, return the key to invalidate
+			if existingMd5 != "" && newMd5 != nil && *newMd5 != existingMd5 {
+				return objectKey, nil
+			}
+
+			return "", nil
+		}).(pulumi.StringOutput)
+
+		p.websiteChangedFileOutputs = append(p.websiteChangedFileOutputs, keyToInvalidate)
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Deploy CDN
+func (p *NitricAzurePulumiProvider) deployCDN(ctx *pulumi.Context) error {
+	profile, err := cdn.NewProfile(ctx, "website-profile", &cdn.ProfileArgs{
+		ResourceGroupName: p.ResourceGroup.Name,
+		Sku: &cdn.SkuArgs{
+			Name: pulumi.String("Standard_Microsoft"),
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Pull the hostname out of the storage-account endpoint.
+	originHostname := p.StorageAccount.PrimaryEndpoints.ApplyT(func(endpoints storage.EndpointsResponse) (string, error) {
+		parsed, err := url.Parse(endpoints.Web)
+		if err != nil {
+			return "", err
+		}
+		return parsed.Hostname(), nil
+	}).(pulumi.StringOutput)
+
+	endpointName := fmt.Sprintf("website-endpoint-%s", p.StackId)
```

Comment:
the `website` part here may not be necessary, Including the stackId here is really good though (should be the same in AWS). possibly something like: `fmt.Sprintf("%s-endpoint", p.StackId)` or the other way around.

Reply:
Made some updates for this.